### PR TITLE
Final SHRD code, plus unrelated bug fixes from Max and Ding

### PR DESCRIPTION
**Pull request by brandenjlynch**

### CODE REVIEW

**Comment by genomaxx** [comment]
The file naming convention for the sHRD functions differs from the rest of the package.
I like the 'shrd_' prefix to those files because it allows for easy reference.
Not sure about the rest having longer titles and more underscores.
Any strong opinions?
The tidyverse style guide (https://style.tidyverse.org/package-files.html), and the R packages book (https://r-pkgs.org/code.html) don't appear to have strong recommendations.

File: R/shrd_levels.R [hunk 1]
```diff
@@ -123,10 +123,10 @@ AssignLevels <- function(segments, thr) {
     n = length(closest_indices)
     for (i in 1:n) {
       # Finding  segment i from `closest_indices` in `segments`
-      segments[which(segments[,1] == closest_indices[i]), 8] = level
+      segments[which((segments[,1] == closest_indices[i])), 8] = level
```
Comment (genomaxx) [inline R/shrd_levels.R:126]: Any reason for the double parentheses? AFAIK that is redundant no?

File: R/shrd_call_lga.R [hunk 2]
```diff
@@ -118,18 +118,18 @@ BreakSmoothToLGA <- function(threshold, segments, granges_obj) {
...
             segments[i+1,4] <- segments[i,4]
-            
+
             gr = GenomicRanges::GRanges(seqnames = c(segments[i+1,2]),
-                                        ranges = GenomicRanges::IRanges(start=c(segments[i+1,4]), end=c(segments[i+1,5])),
+                                        ranges = IRanges::IRanges(start=c(segments[i+1,4]), end=c(segments[i+1,5])),
                                         strand = c("*"))
-            subsetGRobject = GenomicRanges::subsetByOverlaps(granges_obj, gr)
-            
+            subsetGRobject = IRanges::subsetByOverlaps(granges_obj, gr)
```
Comment (genomaxx) [inline R/shrd_call_lga.R:125]: granges_obj?
lol
iranges_obj
minor

File: R/shrd_levels.R (hunk 1 above)
Comment (brandenjlynch) [reply R/shrd_levels.R:126]: Yep, typo.

**Comment by genomaxx** [comment]
I agree that this one should be a merge commit.
Definitely would be nice to rollback easily if necessary.

File: R/shrd_helpers_levels.R [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+  while((length(grep("#",aa))!=0)&&(grep("#",aa)==1)){
+    aa<-scan(seg_file_name,skip=k,nlines=1,what="character",quiet = T)
+    k<-k+1
+  }
+
+  tmp<-read.delim(seg_file_name,skip=k-1, header=T,sep="\t")
+  tmp<-data.matrix(tmp)
+  if(k==2){
+    results<-getInfoSegm(scan(seg_file_name,nlines=1,what="character",quiet = T))
```
Comment (dingma0) [inline R/shrd_helpers_levels.R:63]: should this be GetInfoSegm?

Comment (genomaxx) [reply R/shrd_helpers_levels.R:63]: Looks like it, yup
How did this function run though when testing? o.0

**Comment by genomaxx** [comment]
Broader convo/consideration I think we need to file as an 'Issue':
We are exporting a ton of these shallowHRD functions to the namespace.
Do they all need to user facing? Most depend on one another and are not useful in isolation.
I doubt a user will ever want to touch most of them.
Food for thought.

File: R/shrd_plot_segments.R [hunk 4]
```diff
@@ -0,0 +1,224 @@
+#suppressMessages(library("DescTools"))
+#suppressMessages(library("ggplot2"))
+#suppressMessages(library("gridExtra"))
+#suppressMessages(library("GenomicRanges"))
+#suppressMessages(library("ks"))
+#suppressMessages(library("ggrepel"))
```
Comment (genomaxx) [inline R/shrd_plot_segments.R:6]: leftover code I believe

File: .gitignore [hunk 5]
```diff
@@ -1,3 +1,4 @@
+/shallowHRD
```
Comment (genomaxx) [inline .gitignore:1]: Why is this needed?

File: R/shrd_helpers_levels.R (hunk 3 above)
Comment (brandenjlynch) [reply R/shrd_helpers_levels.R:63]: As discussed, this is part of `ReadSegmFile` which is called by `PrepForLevelsInitialization` on an intermediate file generated during process. However, this part of the function is never actually executed because the if condition isn't satisfied. The creation of the intermediate file is not necessary, so I will create an issue to refactor the code to do everything in-memory.

File: R/shrd_plot_segments.R (hunk 4 above)
Comment (brandenjlynch) [reply R/shrd_plot_segments.R:6]: Yep, this should be removed.

File: .gitignore (hunk 5 above)
Comment (brandenjlynch) [reply .gitignore:1]: It's not -- I think this is a relic from many commits ago when Carlos was working on the package.

**Review by genomaxx — APPROVED** [review]
Looks good for now.
A few issues generated that can be addressed in subsequent PRs.
